### DISCUSSION TITLE
Wechselrichter geht 1- bis 2-mal täglich offline

### DISCUSSION THREAD

**Discussion by BlackFlag69** [hoylabs/OpenDTU-OnBattery#223] [Q&A] [answered]
Einer von 3 Wechselrichtern (HM-1500) wird täglich mehrmals abgeschaltet. In der Ereignisanzeige steht dann "Shutdown by remote control". Ich kann ausschließen, dass es durch ein Script o.ä. geschieht. Es wird immer der erste konfigurierte WR abschaltet. Wenn ich das bemerke starte ich den WR wieder über OpenDTU.

Heute habe ich die Schaltung (ESP32 und NRF24L01+) neu aufgebaut und die Konfiguration nochmal von Hand eingerichtet. Der erste WR ist nun ein anderer. Nachdem ich die WR alle eingerichtet hatte (3 Panel von 4 Panel konfiguriert) lief alles. Als nächstes habe ich einen Stromzähler konfiguriert (SDM72). Dieser wird über Tasmota ausgelesen. Ich lese also die Daten mit HTTP+JSON mit openDTU von Tasmota. Sofort nachdem ich die Einstellungen gespeichert habe, wurde der erste WR abgeschaltet.
Daher vermute ich, dass es einen Seiteneffekt in openDTU gibt, der den ersten WR abschaltet, möglicherweise wenn die Kommunikation (z.B. nach einem Netzwerkproblem) zum Stromzähler neu gestartet wird. Ich habe sonst nichts konfiguriert, kein Power Limiter, kein MQTT, gar nichts.

Kann das jemand nachvollziehen?

**Comment by helgeerbe** [comment]
Kannst du mir den Build Hash nennen (unter Info/System)? Alle Erweiterungen bis auf Power Meter sind aus? Kannst du das Logfile senden?

**Reply by BlackFlag69** [reply]
Das git Hash ist 178d40d. Ich habe das aber seit Beginn vor ca. 2 Wochen mit der damaligen Git-Version.
Ja, alle sonstigen Erweiterungen sind aus.
Ein Log habe ich leider nicht. Du meinst das, was ich bei Info/Konsole sehe und in die Zwischenablage kopieren kann? Logging to syslog wäre jetzt ein nützliches Feature. ;-)

**Comment by BlackFlag69** [comment]
Also ich wette, es hat irgendwas mit dem Verlust der WLAN-Verbindung zu tun, was dann das Abschalten des ersten WR hervorruft.
Um 9:54:50 hat mir mein FHEM eine Nachricht gesandt, dass die Außenkamera nicht mehr erreichbar ist. Um 9:54:56 steht in der Ereignisanzeige des WR, dass er abgeschaltet wurde:

```
09:54:45 | 09:54:45 | 2   | DTU command failed
09:55:16 | 09:55:16 | 2   | DTU command failed
09:54:56 | 09:56:16 | 124 | Shut down by remote control
```

**Comment by IngoEF** [comment]
Weil es ja gerne abgefragt wird:
schreibt openDTU ein Logfile on Board mit und wie kommt man dran?
Oder kriegt man nur eins, wenn es über USB per Monitor in z.B. in PIO erzeugt wird?

**Reply by helgeerbe** [reply]
Ja, detailliertes Log wird über die Serielle-Schnittstelle per USB ausgegeben. Das Konsolen-Log im Webbrowser zeigt leider keine Infos zu Abstürzen oder so.

**Reply by IngoEF** [reply]
Verständlich, ohne OS keine Konsole.

**Comment by IngoEF** [comment]
An das Problem hänge ich mich dran:
openDTU zeigt:
Ereignisanzeige
Begin	Ende	Fehlernummer	Meldung06:34:41	06:34:41	1	Inverter start
06:35:17	06:35:17	2	DTU command failed
09:57:42	10:15:57	124	Shut down by remote control     (keine Ahnung warum)
12:25:17	14:05:13	124	Shut down by remote control     | in dieser Zeit
14:05:21	14:09:10	124	Shut down by remote control     | war dyn. Leistungsregelung an
14:09:17	14:10:02	124	Shut down by remote control     | und das Tasmota Powermeter hat
14:10:03	14:11:21	124	Shut down by remote control     | "Connection refused" warum auch immer
allerdings verstehe ich die Reaktion "Controller Shutdown" in dem Fall nicht.
Dann habe ich die dyn. Leistungsregelung ausgeschaltet, genauso wie die Powermeter Abfrage

ab hier dann
- OpenDTU neugestartet
- Powermeter neu konfiguriert, getestet, funktionierte, gespeichert und
14:37:28	00:00:00	124	Shut down by remote control
das soll ja by Design so sein, aber warum ist dann 
- die Leistungsbegrenzung auf "Aktuelles Limit: 10 W | 2 %"
- WR bleibt aus
?
Das war gestern übrigens auch so.
Konsolenmitschnitt liegt bei.
[Konsole.txt](https://github.com/helgeerbe/OpenDTU-OnBattery/files/11412388/Konsole.txt)

**Comment by helgeerbe** [comment]
Jetzt bin ich ein wenig verwirrt.

1. Änderungen an den Power Meter Einstellugen führen zum reboot des ESP. Das ist so gewollt, da zwischen HTTP, MQTT und seriellem Interface gewechselt werden kann und die Abmeldungen und Initialisierungen im laufenden Betrieb sehr komplex geworden wären. So wird die Config gelesen und das entsprechende Power Meter Interface initialisiert.
2. Power Meter macht keine Änderungen an den Invertern.
3. Nur der PowerLimiter macht Änderungen an den Invertereinstellungen, wenn er aktiviert ist.
4. Der PowerLimiter kann über die Weboberfläche und über MQTT an- und abgeschaltet werden
5. Wenn ihr per MQTT den PowerLimiter schaltet, sollte kein Retain-Flag gesetzt sein. Da, dann immer der PowerLimiter entsprechend der letzten gesendet Nachricht geschaltet wird. 
6. Bitte in der Weboberfläche prüfen, ob der PowerLimiter wirklich aus ist.
7. Wenn der PowerLimiter an ist, und die Werte vom PoweMeter älter als 30s sind, wird das Limit auf 10 W (mögliches Minimum) gesetzt und der Inverter abgeschaltet.

**Comment by helgeerbe** [comment]
@BlackFlag69 ich habe keine Ahnung, warum der Inverter bei dir abgeschaltet werden sollte.
@IngoEF bei dir sehe ich im Logfile: `14:37:12.391 > PL initiated inverter shutdown.` d. h. der PowerLimiter war an. Warum auch immer

**Comment by helgeerbe** [comment]
OK, I guess I found something in WeApi_powerlimiter.cpp line 122

````c++
    config.PowerLimiter_Enabled = root[F("enabled")].as<bool>();
    PowerLimiter.setDisable(false);  // User input clears the PL internal disable flag
````

@MalteSchm why is this flag always set to false? Or why do you introduced this flag anyway. Wouldn't it be better just to use the config value only. In your MQTT Handler you could also change the config value

**Comment by MalteSchm** [comment]
The reason why I did not use config value is that this disables the parameters in the Ui. This seems to be a bad idea as they are no longer accessible in case the Pl is disabled using MQTT

It seemed useful to make sure that the pl is enabled on user input. Hence clearing the disabled flag when calling the WebAPI.

**Comment by IngoEF** [comment]
I can assure you all MQTT (which means tasmota / openDTU and even an ahoy which runs in parallel) switched off after the last desasters I had to go through.
Also -as discribed- Power Limiter is off.
The inverter has been shut down every time I configured Power Meter -> ok, that might be by intention, you already mentioned that before.
I don't understand following:
- missing activation of the inverter automatically after the shutdown
- reduced power limit between 0 to 2% (differs) immediatly after storing the Power Meter activation
- often I had to switch on the inverter up to 5 times to get it working again - which might be a specialiality in my environment (who knows)

**Comment by MalteSchm** [comment]
Ich glaube bei dem hin und her hier können wir auch auf deutsch weiter machen. 
@IngoEF kannst du nochmal das Problem und deine Konfig beschreiben. Das verstehe ich nicht

**Reply by IngoEF** [reply]
Gerne.
Aktuell sind MQTT- und Dyn. Power Limiter desaktiviert.
Es geht um Aktivierung und/oder Änderungen in der PowerMeter Konfiguration.
Nach dem Speichern geht der WR aus. Das mag gewollt sein, allein einsehen kann ich das nicht.
Er muss dann manuell wieder angeschaltet werden. Warum geht das nicht automatisch?
Sobald er dann eingeschaltet wurde, steht Powerlimit bei "Aktuelles Limit: 10 W | 2 %" und muss von Hand wieder auf (bei mir) 100% eingestellt werden.
Warum fasst die Firmware das Leistungslimit überhaupt an?
Hier hängt jetzt ein Logfile für den Vorgang "Timeout für HTTP/JSON-Request von 1500 auf 2000 ms erhöhen" dran.
[platformio-device-monitor-230507-151201.log](https://github.com/helgeerbe/OpenDTU-OnBattery/files/11415088/platformio-device-monitor-230507-151201.log)

Noch ein Nachsatz:
während ich dies hier schrieb, ging der Wechselrichter ohne mein Einwirken wieder aus und das Limit wird genauso reduziert angezeigt wie oben gezeigt, jetzt leider ohne Logfile, weil USB-Konsole schon beendet.

**Reply by helgeerbe** [reply]
@IngoEF Ich habe eben den PowerLimiter bei mir ausgeschaltet. Am PowerMeter den Timeout geändert. Der ESP startet neu und der Inverter läuft mit den vorherigen Werten. Da geht nichts aus.. Wenn bei dir der Wechselrichter aus ist und das Limit auf auf 10W steht, dann muss aus irgendeinem Grund der PowerLimiter aktiv sein. Nur ich kann es in deinem Logfile nicht sehen, dass der PowerLimiter irgendetwas zu regeln versucht.

**Reply by MalteSchm** [reply]
Ok ich habe es nachvollziehen können.

Das Problem ist folgendes: 

Die Änderung an den Power Meter Werten führt zu einem Restart des ESP. Der Powerlimiter wird daraufhin erst mal mit dem active state initialisiert. (PowerLimiter.h ->     plStates _plState = ACTIVE; ) 
Da der PL aber disabled ist schaltet der PL den Inverter ab.

Fix sollte nicht schwer sein. Moment...

**Reply by MalteSchm** [reply, marked as answer]
@helgeerbe PR is here: https://github.com/helgeerbe/OpenDTU-OnBattery/pull/224
Briefly tested

Ich denke das das auch das Problem von @BlackFlag69 adressiert. Hier war ja der PL auch aus.

**Reply by IngoEF** [reply]
@MalteSchm @helgeerbe 
Wenn's gefunden wurde, ist wieder ein Schritt getan. Im Moment habe ich nicht die Zeit, das zu testen.
Mache ich morgen und berichte.

**Reply by BlackFlag69** [reply]
Toll, das ihr da so intensiv dran geblieben seid. Ich teste das bei mir auch gleich morgen und gebe bescheid, ob bei mir die shut Downs weg sind.

**Reply by IngoEF** [reply]
Erfolg!
Problem erledigt.
Super gemacht, dankeschön.
Ich weiß jetzt gerade nicht (falls mir das überhaupt möglich sein sollte), welche ich als Antwort markieren könnte.

**Reply by helgeerbe** [reply]
Danke für die Rückmeldung. Ich habe den PR von Malte markiert. Werde ich gleich in Development mergen.
